### DISCUSSION TITLE
reference-doc in yaml not being picked up by pandoc

### DISCUSSION THREAD

**Discussion by carljoseph** [jgm/pandoc#9722] [General]
I have a markdown file with some yaml arguments at the top

```
---
bibliography: /Users/me/bibliography.bib
csl: /Users/me/chicago-fullnote-bibliography.csl
reference-doc: /Users/me/template.docx
---
# My markdown doc
```

When I run the following command, it doesn't seem to pick up the `reference-doc`. But when I mention it explicitly on the command line it works.

**Doesn't work**
```
pandoc -s -f markdown --citeproc --to docx --verbose --output temp.docx myinput.md
[INFO] Running filter citeproc
[INFO] Loaded /Users/me/chicago-fullnote-bibliography.csl from /Users/me/chicago-fullnote-bibliography.csl
[INFO] Loaded /Users/me/bibliography.bib from /Users/me/bibliography.bib
[INFO] Completed filter citeproc in 748 ms
```
_Note, the lack of the template.docx above_

**Works**
```
pandoc -s -f markdown --citeproc --to docx --verbose --reference-doc=/Users/me/template.docx --output temp.docx myinput.md
[INFO] Running filter citeproc
[INFO] Loaded /Users/me/chicago-fullnote-bibliography.csl from /Users/me/chicago-fullnote-bibliography.csl
[INFO] Loaded /Users/me/bibliography.bib from /Users/me/bibliography.bib
[INFO] Completed filter citeproc in 755 ms
[INFO] Loaded /Users/me/template.docx from /Users/me/template.docx
```

Any ideas?

**Comment by fiapps** [comment]
> I have a markdown file with some yaml arguments at the top

This is actually what pandoc calls a [YAML metadata block](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block). The distinction is important, because it can only provide values for metadata, not all arguments. If you want to use YAML to set a reference document, you need to use a [defaults file](https://pandoc.org/MANUAL.html#defaults-files).

**Reply by carljoseph** [reply]
~Thanks for this. Is there a reference of what can / can't be set through the metadata block?~ Nevermind, I found them [here](https://pandoc.org/MANUAL.html#metadata-variables)
